docs/rfcs/0001-congratulations-letter-to-dashboard.md

```diff
@@ -0,0 +1,39 @@
+## Add Congratulations Letter to Dashboard
+
+### Abstract
+
+As a learner who has earned a program certificate, I sometimes have trouble explaining the meaning and value of the certificate. It's useful to have a letter from the faculty that describes my accomplishment that I can share with employers, in addition to the certificate.
+
+### Architecture Changes
+
+We have already codebase for program certificate creation, logic of creating program congratulation is more or less same as program certificate creation.
+
+#### Database:
+We can add new model `MicromastersProgramCompletion` in `grades/models.py` with the following fields.
+
+* User
+* Course
+* Hash/UUID
+
+Previously we are using manually created hash for program certificates in `MicromastersProgramCertificate`. There is a open question about using UUID.
+
+**Question:** Can we use [uuid](https://docs.djangoproject.com/en/2.1/ref/models/fields/#uuidfield) to uniquely identify user's earned letter. With the usage of it we would no longer need to override save() method to create md5 hash.
+
+#### Business logic:
+**New certificates:** We will add new signal in `grades/signals.py`, which will triggered whenever there is a new creation in course certificate. There will be a method, similar to `grades.api.generate_program_certificate` in which we can filter non-fa users and create letter for related program if available.
```

Comment:
Do you mean to add a signal to program certificate?

Reply:
The existing signal for generating a program cert is based on a new course certificate being created: `grades.signals.handle_create_coursecertificate`. I think it makes sense to add logic to the existing `grades.api.generate_program_certificate` task and rename it, or just trigger a new task after the `generate_program_certificate` task in the signal handler

Reply:
I just want to separate both methods, may be in future we want more checks around it.

Reply:
@pdpinch thoughts ?

Reply:
I think it makes sense for the signals to be separate, as long as their isn't too much code (and test) duplication.